crates/wasm/src/compile.rs

```diff
@@ -58,11 +58,32 @@ impl Default for WASMCompileOptions {
     }
 }
 
-fn add_noir_lib(context: &mut Context, crate_name: &str) {
-    let path_to_lib = Path::new(&crate_name).join("lib.nr");
-    let library_crate = prepare_dependency(context, &path_to_lib);
-
-    propagate_dep(context, library_crate, &crate_name.parse().unwrap());
+fn add_noir_lib(context: &mut Context, library_name: &str) {
+    let path_to_lib = Path::new(&library_name).join("lib.nr");
+    let library_crate_id = prepare_dependency(context, &path_to_lib);
+
+    add_dep(context, *context.root_crate_id(), library_crate_id, library_name.parse().unwrap());
+
+    // TODO: Remove this code that attaches every crate to every other crate as a dependency
+    let root_crate_id = context.root_crate_id();
+    let stdlib_crate_id = context.stdlib_crate_id();
+    let other_crate_ids: Vec<_> = context
+        .crate_graph
+        .iter_keys()
+        .filter(|crate_id| {
+            // We don't want to attach this crate to itself or stdlib, nor re-attach it to the root crate
+            crate_id != &library_crate_id
+                && crate_id != root_crate_id
+                && crate_id != stdlib_crate_id
+        })
+        .collect();
+
+    for crate_id in other_crate_ids {
+        context
+            .crate_graph
+            .add_dep(crate_id, library_name.parse().unwrap(), library_crate_id)
+            .expect(&format!("ICE: Cyclic error triggered by {} library", library_name));
+    }
```

Comment:
This is a more precise version of `propagate_dep` because it filters the root and stdlib crate ids. The stdlib should never have deps appended and the root will already have the library because of `add_dep` above.